ts/routes/image-occlusion/shapes/to-cloze.ts

```diff
@@ -166,7 +168,7 @@ function shapeOrShapesToCloze(
 
     // Maintain existing ordinal in editing mode
     let ordinal = shapeOrShapes.ordinal;
-    if (ordinal === undefined) {
+    if (ordinal === undefined || get(groupItemsModified)) {
```

Comment:
This can break the ordinals of ungrouped masks when a grouping operation is done on other masks in edit mode.

Can we instead update the ordinals of grouped objects when they are grouped?
For example, If we group objects with the ordinals 2 and 4, all objects in the group will be assigned a single ordinal (maybe the smallest one, so 2).

Reply:
I don't think so, because let's say a user have created group cloze with more than two shapes as single group cloze, then during review he edited the shape to ungroup to individual shapes. Now total cards increased from one to total number of shapes. Now again during review he split it into two group, now ordinal must start from beginning to have consistent result, (Card 1 and Card 2)

Reply:
I don't think we can solve this for all situations, especially if the user is closing and opening the editor between changes. I'd assumed that if we give new ordinals to shapes when they're removed from a group, that would be sufficient?

Reply:
The new ordinal to removed shapes may not also solve this, I have manually assigned cloze number to removed shape by editing clozes in Occlusion field and result to that Card was something like this. It makes all shapes as occlusion for front as well as back.

The other solution should be when shapes are merged, we can merge cards also, but we cannot know which card associated with which shape. So, I can say that we should warning message when users performing group/ungroup of shapes.

![image](https://github.com/ankitects/anki/assets/12841290/2474f8c3-8a51-4abc-a07c-64c992c31c04)

Reply:
We could show a warning when the group/ungroup tools are used on an existing note in the future, but I'm not too concerned about it at the moment as long as it's not completely breaking.

Reply:
Just to check we're on the same page - would you like this PR merged in without the changes that Abdo suggested?

Reply:
I have thought of implementation idea for Abdo suggestion, I have implemented it in next commit, and it seems, it also solves the empty card issues. But it still needs to some review.

Reply:
Users need to do `Tools -> Empty Cards`, I have tried to solve but once cards added, the ordinal decreases but Cards remains same, so it needs to be fixed separately.

Reply:
Sorry, I'm not quite following - what do you think needs to be done differently?

Reply:
Nothing, I was implementing different solution when ungroup shapes. In case of ungroup shapes, find all the ordinal from shapes excluding groups shapes then start loop from beginning and put the ordinal to shapes in order. But I implemented simplest solution, use max ordinal (increment if required).